File: Génération_Code_Application/Codex5.3_VistaCanvas/frontend/src/types/review.ts
export interface Review {
  id: number;
  product_id: number;
  user_id: number;
  username: string;
  rating: number;
  title: string;
  body: string;
  image_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface ReviewInput {
  rating: number;
  title: string;
  body: string;
  image_url?: string | null;
}

File: Génération_Code_Application/GPT5.5_VoltStream/backend/app/schemas/review.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class ReviewRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    user_id: int
    user_name: str
    rating: int
    title: str
    body: str
    image_url: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class ReviewUpdate(BaseModel):
    rating: Optional[int] = Field(None, ge=1, le=5)
    title: Optional[str] = Field(None, min_length=1, max_length=120)
    body: Optional[str] = Field(None, min_length=10, max_length=5000)
    image_url: Optional[str] = Field(None, max_length=500)
